connectors/tests/test_byoc.py

```diff
@@ -213,70 +212,54 @@ def test_utc():
-
-    expected_filtering = {
-        "advanced_snippet": {
-            # "value" should be omitted by extracting content inside "value" and moving it one level up
-            "find": {"settings": {}}
-        },
+    job_filtering = {
+        "advanced_snippet": ACTIVE_ADVANCED_SNIPPET,
         "rules": [{"id": ACTIVE_RULE_ONE_ID}, {"id": ACTIVE_RULE_TWO_ID}],
         "validation": FILTERING_VALIDATION_VALID,
     }
 
-    job = SyncJob(connector_id="connector-id", elastic_index=jobs_index)
-
-    headers = {"X-Elastic-Product": "Elasticsearch"}
-    mock_responses.post(
-        "http://nowhere.com:9200/.elastic-connectors/_refresh", headers=headers
-    )
+    jobs_index = SyncJobIndex(elastic_config=config)
 
-    sent_docs = []
+    index_mock = AsyncMock()
+    update_mock = AsyncMock()
 
-    def callback(url, **kwargs):
-        sent_docs.append(json.loads(kwargs["data"]))
-        return CallbackResult(
-            body=json.dumps({"_id": "1"}), status=200, headers=headers
-        )
+    jobs_index.client.index = index_mock
+    jobs_index.client.update = update_mock
 
-    mock_responses.post(
-        "http://nowhere.com:9200/.elastic-connectors-sync-jobs/_doc",
-        callback=callback,
-        headers=headers,
-    )
+    job = SyncJob(connector_id="connector-id", elastic_index=jobs_index)
 
-    mock_responses.put(
-        "http://nowhere.com:9200/.elastic-connectors-sync-jobs/_doc/1",
-        callback=callback,
-        headers=headers,
-    )
-    mock_responses.post(
-        "http://nowhere.com:9200/.elastic-connectors-sync-jobs/_update/1",
-        callback=callback,
-        repeat=True,
-    )
+    await job.start(filtering=job_filtering)
 
     assert job.duration == -1
-    await job.start(filtering=ACTIVE_FILTERING_DEFAULT_DOMAIN)
     assert job.status == JobStatus.IN_PROGRESS
     assert job.job_id is not None
-    await asyncio.sleep(0.2)
+
+    job_def = index_mock.call_args.kwargs["document"]
+
+    expected_filtering = {
+        "advanced_snippet": {
+            # "value" should be omitted by extracting content inside "value" and moving it one level up
+            "find": {"settings": {}}
+        },
+        "rules": [{"id": ACTIVE_RULE_ONE_ID}, {"id": ACTIVE_RULE_TWO_ID}],
+        "validation": {"state": "valid", "errors": []},
+    }
+
+    assert job_def["status"] == "in_progress"
+    assert job_def["connector"]["filtering"] == expected_filtering
+
     await job.done(12, 34)
+
     assert job.status == JobStatus.COMPLETED
-    await client.close()
-    assert job.duration >= 0.2
-
-    # verify what was sent
-    assert len(sent_docs) == 2
-    doc, update = sent_docs
-    assert doc["status"] == "in_progress"
-    assert doc["connector"]["filtering"] == expected_filtering
-    assert update["doc"]["status"] == "completed"
-    assert update["doc"]["indexed_document_count"] == 12
-    assert update["doc"]["deleted_document_count"] == 34
+    assert job.duration > -1
```

Comment:
Hmmm, but can it actually be in the range (-1; 0]. If so, how come?

Reply:
Oh wow of course you're right, was so focused on tracking down, why we use microseconds and if it has implications somewhere else. Adjusted it!